// Copyright 2023 Intrinsic Innovation LLC

syntax = "proto3";

package intrinsic_proto.icon;

import "intrinsic/icon/equipment/force_control_settings.proto";

message Icon2PositionPart {
  string part_name = 1;
  string world_robot_collection_name = 3;
  reserved 2;
  reserved "connection_config";
}

message Icon2TorquePart {
  string part_name = 1;
  string world_robot_collection_name = 3;
  reserved 2;
  reserved "connection_config";
}

message Icon2GripperPart {
  string part_name = 1;
  string world_robot_collection_name = 3;
  reserved 2;
  reserved "connection_config";
}

message Icon2AdioPart {
  message IconTarget {
    string part_name = 1;
    string world_robot_collection_name = 3;
    reserved 2;
    reserved "connection_config";
  }

  oneof target {
    IconTarget icon_target = 3;
  }
}

message Icon2ForceTorqueSensorPart {
  string part_name = 1;
  string world_robot_collection_name = 3;
  reserved 2;
  reserved "connection_config";
  // Parameters used for all force control skills.
  ForceControlSettings force_control_settings = 4;
}

message Icon2RangefinderPart {
  string part_name = 1;
  string world_robot_collection_name = 3;
  reserved 2;
  reserved "connection_config";
}
